### PR TITLE
`Team{Thread,Vector}MDRange` and `ThreadVectorMDRange` convert to rst

### PR DESCRIPTION
**Pull request by fnrizzi**
self-explanatory (somehow this was missed)

# `TeamThreadMDRange` 

|  before  |  after  |
|  -----    | ------ |
| image | image |

# `TeamVectorMDRange` 

|  before  |  after  |
|  -----    | ------ |
| image | image |

# `ThreadVectorMDRange ` 

|  before  |  after  |
|  -----    | ------ |
| image | image |

### CODE REVIEW

**Comment by ajpowelsnl** [comment]
@fnrizzi -- in an effort to move towards a standard for API entries we decided on these main sections:

- Name
- Header file
- Description
- Interface
- Parameters
- Constructors (where relevant)
- Methods
- Example

Nic's PR summarizes this standard:

[TEMPLATE FOR API ENTRY](https://github.com/kokkos/kokkos-core-wiki/pull/385/files)

Would it be possible to re-work this PR for the new standard?

File: docs/source/API/core/policies/TeamVectorMDRange.rst [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+   * Every member thread of ``team`` must call the operation in the same branch, i.e. it is not legal to have some
+       threads call this function in one branch, and the other threads of ``team`` call it in another branch.
+
+   * ``N >= 2 && N <= 8`` is true;
+
+Examples
+--------
+
+.. code-block:: cpp
+
```
Comment (ajpowelsnl) [inline docs/source/API/core/policies/TeamVectorMDRange.rst:55]: Move general form from above to the beginning of the Example as a comment.

File: docs/source/API/core/policies/ThreadVectorMDRange.rst [hunk 2]
```diff
@@ -1,52 +1,60 @@
...
 
-Header File: `Kokkos_Core.hpp`
+.. role::cpp(code)
+    :language: cpp
+
+Header File: ``<Kokkos_Core.hpp>``
+
+Usage
+-----
+
```
Comment (ajpowelsnl) [inline docs/source/API/core/policies/ThreadVectorMDRange.rst:11]: Please change `Usage` to `Description` and add a couple of explanatory sentences.

File: docs/source/API/core/policies/ThreadVectorMDRange.rst [hunk 3]
```diff
@@ -1,52 +1,60 @@
...
+.. code-block:: cpp
 
-Usage: 
-  ```c++
     parallel_for(ThreadVectorMDRange<Kokkos::Rank<...>, TeamHandle>(team, extent1, extent2, ...),
       [=] (int i1, int i2, ...) {...});
-    parallel_reduce(ThreadVectorMDRange<Kokkos::Rank<...>, TeamHandle>(team, extent1, extent2, ...), 
+    parallel_reduce(ThreadVectorMDRange<Kokkos::Rank<...>, TeamHandle>(team, extent1, extent2, ...),
       [=] (int i1, int i2, ..., double& lsum) {...}, sum);
-  ```
```
Comment (ajpowelsnl) [inline docs/source/API/core/policies/ThreadVectorMDRange.rst:11]: Move the general form to the beginning of the Example as a comment.

File: docs/source/API/core/policies/ThreadVectorMDRange.rst [hunk 4]
```diff
@@ -1,52 +1,60 @@
...
-   struct ThreadVectorMDRange<Rank<N, ...>, TeamHandle> {
+   struct ThreadVectorMDRange<Rank<N, ...>, TeamHandle>
+   {
      ThreadVectorMDRange(team, extent1, extent2, ..., extentN) { /* ... */ }
-
-     /* ... */
    };
-  ```
 
-## Description
```
Comment (ajpowelsnl) [inline docs/source/API/core/policies/ThreadVectorMDRange.rst:26]: Are lines 33 -34 an overview of what `ThreadVectorMDRange` does?  If so please see comment above about `Usage`.  Following the recently-adopted standard for API entries, `Arguments` would become `Parameters`, and `Requirements` would be expressed as required parameters (vs. optional).

**Review by ajpowelsnl — CHANGES_REQUESTED** [review]

**Review by ldh4 — APPROVED** [review]

**Review by masterleinad — APPROVED** [review]
Looks OK to me.

**Review by crtrott — APPROVED** [review]

**Pull request merged**